server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -804,7 +804,7 @@ public void executeFetchPhase(ShardFetchRequest request, SearchShardTask task, A
                 }
                 searchContext.assignRescoreDocIds(readerContext.getRescoreDocIds(request.getRescoreDocIds()));
                 searchContext.searcher().setAggregatedDfs(readerContext.getAggregatedDfs(request.getAggregatedDfs()));
-                searchContext.docIdsToLoad(request.docIds(), request.docIdsSize());
```

Comment:
I had to double check this - but it looks like we really were always using the size of the array. weird.

Reply:
Yep, took me by surprise as well